### DISCUSSION TITLE
Hope to be able to add fingerprint information

### DISCUSSION THREAD

**Discussion by zoushucai** [encode/httpx#2827] [General] [answered]
Recently, I encountered some issues while using the `httpx` package. It was later discovered that the website had fingerprint verification, and I used the `curl_cffi` package to solve the problem. Can the author also introduce the mechanism of the `curl_cffi` package, Thank you

**Comment by zanieb** [comment]
Hi! We request that issues start as discussions here.

Can you explain the issues that you encountered?

**Comment by zoushucai** [comment]
Hello, this is a typical example from [https://github.com/yifeikong/curl_cffi](https://github.com/yifeikong/curl_cffi).  If we access it through a browser, we can get the value of '"akamai_hash', but if it is a crawler, the corresponding result cannot be obtained,

**Comment by zoushucai** [comment]
The code is as follows:
```python
import httpx
client = httpx.Client()
rep = client.get("https://tls.browserleaks.com/json")
print(rep.json())
```
The result is that the following field values are empty

```
'akamai_hash': '', 'akamai_text': ''
```
In fact, if you use a browser to access it, these two values do exist

**Reply by My-Responsitories** [reply, marked as answer]
these only available in http/2
```python
import httpx
client = httpx.Client(http2=True)
rep = client.get("https://tls.browserleaks.com/json")
print(bool(rep.json()['akamai_hash']))
# True
```

**Comment by zoushucai** [comment]
@My-Responsitories Wow, it's incredibly astonishing. I didn't realize how powerful HTTP2 is here. Thank you for opening the door to a new world for me. Previously, I was only using direct GET requests.

**Comment by zoushucai** [comment]
Thanks
